CactbotOverlay/FFXIVProcess.cs

```diff
@@ -570,525 +504,445 @@ public class EntityData {
-      byte[] bytes = GetJobSpecificData();
-      if (bytes == null)
-        return null;
+    [Serializable]
+    [StructLayout(LayoutKind.Explicit)]
+    public struct DragoonJobMemory {
+      [NonSerialized]
+      [FieldOffset(0x00)]
+      private ushort blood_or_life_ms;
 
-      var j = new BardJobData();
-      j.song_ms = BitConverter.ToUInt16(bytes, kJobDataInnerStructOffsetJobSpecificData);
-      // When the song time is 0, the other fields are not well defined and may be left
-      // in an incorrect state.
-      if (j.song_ms > 0) {
-        j.song_procs = bytes[kJobDataInnerStructOffsetJobSpecificData + 2];
-        j.song_type = (BardJobData.Song)bytes[kJobDataInnerStructOffsetJobSpecificData + 3];
-      }
-      return j;
-    }
+      [NonSerialized]
+      [FieldOffset(0x02)]
+      private byte stance; // 0 = None, 1 = Blood, 2 = Life
 
-    public class DragoonJobData {
-      public uint blood_ms = 0;
-      public uint life_ms = 0;
-      public uint eyes_amount = 0;
+      [FieldOffset(0x03)]
+      public byte eyesAmount;
 
-      public override bool Equals(object obj) {
-        var o = obj as DragoonJobData;
-        return o != null &&
-          blood_ms == o.blood_ms &&
-          life_ms == o.life_ms &&
-          eyes_amount == o.eyes_amount;
+      public uint bloodMilliseconds {
+        get {
+          if (stance == 1)
+            return blood_or_life_ms;
+          else
+            return 0;
+        }
       }
-
-      public override int GetHashCode() {
-        int hash = 17;
-        hash = hash * 31 + blood_ms.GetHashCode();
-        hash = hash * 31 + life_ms.GetHashCode();
-        hash = hash * 31 + eyes_amount.GetHashCode();
-        return hash;
+      public uint lifeMilliseconds {
+        get {
+          if (stance == 2)
+            return blood_or_life_ms;
+          else
+            return 0;
+        }
       }
-    }
+    };
 
-    public DragoonJobData GetDragoon() {
-      byte[] bytes = GetJobSpecificData();
-      if (bytes == null)
-        return null;
+    [Serializable]
+    [StructLayout(LayoutKind.Explicit)]
+    public struct NinjaJobMemory {
+      [FieldOffset(0x00)]
+      public uint hutonMilliseconds;
 
-      var j = new DragoonJobData();
-      j.blood_ms = j.life_ms = 0;
-      byte stance = bytes[kJobDataInnerStructOffsetJobSpecificData + 2];
-      if (stance == 1) {
-        j.blood_ms = BitConverter.ToUInt16(bytes, kJobDataInnerStructOffsetJobSpecificData);
-      } else if (stance == 2) {
-        j.life_ms = BitConverter.ToUInt16(bytes, kJobDataInnerStructOffsetJobSpecificData); ;
-      }
-      j.eyes_amount = bytes[kJobDataInnerStructOffsetJobSpecificData + 3];
-      return j;
-    }
+      [FieldOffset(0x04)]
+      public byte ninkiAmount;
 
-    public class NinjaJobData {
-      public uint huton_ms = 0;
-      public uint ninki_amount = 0;
+      [FieldOffset(0x05)]
+      public byte hutonCount; //TODO: Confirm this.
```

Comment:
It sure seems like this is true, but what in the world???  I would say it's probably better to not expose this, as it's not useful.

Reply:
Yeah I couldn't believe it was actually counting that. I guess it's an artefact from the numerous reworks ninja has had..